DemoAppSwiftUI/AppDelegate.swift

```diff
@@ -57,10 +57,14 @@ class AppDelegate: NSObject, UIApplicationDelegate {
         
         DispatchQueue.main.asyncAfter(deadline: .now() + 2.5) {
             withAnimation {
-                AppState.shared.userState = .notLoggedIn
+                if AppState.shared.userState == .launchAnimation {
+                    AppState.shared.userState = .notLoggedIn
+                }
```

Comment:
We need this because if a push notification is the way to enter the app we don't want to change the state, right?

Reply:
Yes, exactly. There's automatic login if the app is closed.